applications/lfrfid/helpers/decoder_ioprox.cpp

```diff
@@ -0,0 +1,95 @@
+#include "decoder_ioprox.h"
+#include <furi_hal.h>
+#include <cli/cli.h>
+#include <utility>
+
+constexpr uint32_t clocks_in_us = 64;
+
+constexpr uint32_t jitter_time_us = 20;
+constexpr uint32_t min_time_us = 64;
+constexpr uint32_t max_time_us = 80;
+
+constexpr uint32_t min_time = (min_time_us - jitter_time_us) * clocks_in_us;
+constexpr uint32_t mid_time = ((max_time_us - min_time_us) / 2 + min_time_us) * clocks_in_us;
+constexpr uint32_t max_time = (max_time_us + jitter_time_us) * clocks_in_us;
+
+constexpr uint32_t baud_time = 32000;
+
+bool DecoderIoProx::read(uint8_t* data, uint8_t data_size) {
+    bool result = false;
+    furi_assert(data_size >= 4);
+
+    if(ready) {
+        result = true;
+        ioprox.decode(raw_data, sizeof(raw_data), data, data_size);
+        ready = false;
+    }
+
+    return result;
+}
+
+void DecoderIoProx::process_front(bool is_rising_edge, uint32_t time) {
+    if(ready) {
+        return;
+    }
+
+    // Always track the time that's gone by.
+    current_period_duration += time;
+    demodulation_sample_duration += time;
+
+    // If a baud time has elapsed, capture a sample.
+    if(demodulation_sample_duration >= baud_time) {
+        store_data(current_demodulated_value);
+        demodulation_sample_duration = 0;
+    }
```

Comment:
Picks up everything and theoretically leads to false positive because we don't check if it's actually FSK modulation like in DecoderHID26 (it's the same modulation as far as I know).

Reply:
I can see how this would be an issue in terms of wasting CPU time in the detection loop. (It wouldn't lead to false positives unless the thing that was picked up happened to both have pulse trains with the correct periods _and_ managed a correct packet checksum.)

I've adjusted it so values are only stored if there's valid FSK, so should be fixed.

> (it's the same modulation as far as I know).

This is the same modulation as HID26; but with different sample-timing requirements / a different baud-rate. This *should* hopefully be equivalent, but with a slightly more correct clock-recovery behavior, and the ability to tweak the baud clock by changing a single const. 

(I don't think the slight increase in clock-recovery correctness really actually _matters_, here; but it's a bit nicer given the ioProx tags have way lower transition density than e.g. most HID tags.)